.github/DEVELOPMENT.md

```diff
@@ -170,41 +178,85 @@ Enable errorprone ([Error Prone Installation#IDEA](https://errorprone.info/docs/
+The Trino Web UI is composed of several React components and is written in JSX
+and ES6. This source code is compiled and packaged into browser-compatible
+Javascript, which is then checked in to the Trino source code (in the `dist`
+folder). You must have [Node.js](https://nodejs.org/en/download/) and
+[Yarn](https://yarnpkg.com/en/) installed to execute these commands. To update
+this folder after making changes, simply run:
 
     yarn --cwd core/trino-main/src/main/resources/webapp/src install
 
-If no Javascript dependencies have changed (i.e., no changes to `package.json`), it is faster to run:
+If no Javascript dependencies have changed (i.e., no changes to `package.json`),
+it is faster to run:
 
     yarn --cwd core/trino-main/src/main/resources/webapp/src run package
 
-To simplify iteration, you can also run in `watch` mode, which automatically re-compiles when changes to source files are detected:
+To simplify iteration, you can also run in `watch` mode, which automatically
+re-compiles when changes to source files are detected:
 
     yarn --cwd core/trino-main/src/main/resources/webapp/src run watch
 
-To iterate quickly, simply re-build the project in IntelliJ after packaging is complete. Project resources will be hot-reloaded and changes are reflected on browser refresh.
+To iterate quickly, simply re-build the project in IntelliJ after packaging is
+complete. Project resources will be hot-reloaded and changes are reflected on
+browser refresh.
+
+## Releases
+
+Trino aims for frequent releases, generally once per week. This is a goal but
+not a guarantee, as critical bugs may lead to a release being pushed back or
+require an extra emergency release as a patch.
+
+At the start of each release cycle, a GitHub issue is filed and pinned to track
+all necessary release notes. For example, see [the issue for Trino 395](https://github.com/trinodb/trino/issues/13913).
+In addition, a release notes pull request is updated and maintained throughout
+the week, tracking all merged commits to ensure every change is properly
+documented and noted. This uses the [release note template](../docs/release-template.md),
+with changes in each section arranged to have new features first, performance
+improvements second, and bugfixes third. See [the release notes for 395](https://github.com/trinodb/trino/pull/13975)
+as an example.
+
+Once it is time to release, the release process is kicked off. A code freeze is
+announced on the Trino Slack in the #releases channel, and then the
+[release scripts](https://github.com/trinodb/release-scripts) and the Maven
+release plugin are utilized to update all necessary files and update Trino to
+the next version.
```

Comment:
This is basically word vomit and very hand-wavey. @mosabua let me know if you can go into more detail here, I actually have no clue how a release happens 😆

Reply:
I think this is good enough .. could even be cut down to "and then a maintainer performs the  release"